core/src/hir/lowering.rs

```diff
@@ -1073,14 +1189,50 @@ impl<'ast> LoweringContext<'ast> {
                     Err(())
                 }
             }
-            ast::CustomType::Enum(enm) => {
-                let tcx_id = self.lookup_id.resolve_enum(enm).expect("enum is in env");
+            ast::CustomNamedConstruct::CustomType(ast::CustomType::Enum(enm)) => {
+                let tcx_id = self.lookup_id.resolve_enum(&enm).expect("enum is in env");
 
                 Ok((
                     ParamSelf::new(SelfType::Enum(EnumPath::new(tcx_id))),
                     self_param_ltl.no_self_ref(),
                 ))
             }
+            ast::CustomNamedConstruct::Trait(trt) => {
```

Comment:
issue: not clear to me that `lower_param_self` needs to deal with traits at all

Reply:
I had thought that trait methods that take a `self` param would need this

Reply:
^ nevermind, you're right (re: offline discussion with @mmaurer); I'm instead adding a different kind of `self` reference for trait methods